flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/DeclarativeSlotManagerTest.java

```diff
@@ -1424,6 +1428,37 @@ public void testReclaimInactiveSlotsOnClearRequirements() throws Exception {
         }
     }
 
+    @Test
+    public void testProcessResourceRequirementsWithDelay() throws Exception {
```

Comment:
We'd better also test that the check method will be triggered only once in one delay period.

Reply:
Verified by new added

```
Assertions.assertEquals(1, allocatedResourceCounter.get());
```

Reply:
I think we can process another resource requirement before manually triggering the scheduled task. Then the number of the pending task should be still equal to 1.

Reply:
Yes, updated